app.ts

```diff
@@ -0,0 +1,216 @@
+            for (let j = 0; j < numColumns; j++){
+                let td = document.createElement("td");
+                if (dataSize > i)
+                    td.appendChild(document.createTextNode(data[i][j]));
+                tr.appendChild(td);
+            }
+            tbdy.appendChild(tr);
+        }
+        table.appendChild(tbdy);
+        canvas.appendChild(table);
+    }
+
+    /**
+     * Finds the number of entries within a data set.
+     * @param data An array of data, type is unknown (any).
+     */
+    function getSize(data: any) {
+        let i = 0;
+        let entry;
+        for (entry in data) {
+            i++;
+        }
+        return i;
+    }
+
+    /**
+     * Sends out an HTTP request to retrieve data records, coupled with generating a table to display the records.
+     * @param from The ID value from which to start searching.
+     * @param to The ID value from which to stop searching.
+     */
+    function getRecords(from: number, to: number) {
+        $.ajax({
+            url: "http://localhost:2050/records",
+            data: {from: from, to: to},
+            success: function (result) {
+                let data = JSON.parse(result);
+                generateTable(data, columns);
+            },
+            error: function (err) {
+                $("body").text("Error: " + err.status + " " + err.statusText);
+            }
+        });
+    }
+
+    /**
+     * Generates and returns a "thead" object containing column headings.
+     * @param columns An object array containing the column heading values.
+     */
+    function generateHeadings(columns: any) {
+        let thead = document.createElement("thead");
+        let tr = document.createElement("tr");
+        for (let j = 0; j < numColumns; j++) {
+            let td = document.createElement("td");
+            td.appendChild(document.createTextNode(columns[j]));
+            tr.appendChild(td);
+        }
+        thead.appendChild(tr);
+        return thead;
+    }
+
+    /**
+     * Validates that the field value is a number.
+     * @param from The ID value from which to start searching.
+     */
+    function validate(from: number) {
+        if (isNaN(from)) {
+            alert("\"From\" field does not have a number value.");
+            return false;
+        } else if (from < 0) {
+            alert("\"From\" value cannot be negative.");
+            return false;
+        }  else if (from > recordCount-1) {
+            alert("\"From\" value exceeds the record count.");
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Adjust the number of records shown in the table according to the window size.
+     */
+    function adjustTableRecordCount() {
+        let height = window.innerHeight;
+        let fontSize = getComputedStyle(document.documentElement).fontSize + "";
+        let rowHeight = parseFloat(fontSize)*2.5;
+        if (rowHeight !== undefined){
```

Comment:
rather do something like
```ts 
    if (typeof rowHeight !== 'undefined') {
        // Implementation
    }
```
But as a secondary aside, @FritzOnFire will probably end up telling you to invert the conditional so that you can get rid of the unnecessary indentation in the code by doing something like
```ts
    if (typeof rowHeight === 'undefined') {
        return;
    }
``` 
and then continuing with the implementation after this single conditional.